go/enclave/events/subscription_manager.go

```diff
@@ -126,7 +135,7 @@ func (s *SubscriptionManager) FilteredSubscribedLogs(logs []*types.Log, rollupHa
 		rollupNumber := logItem.BlockNumber
 
 		// We check whether the log is relevant to each subscription.
-		// TODO - #1016 - This can blow up if a subscription is added while we are iterating over the subscriptions.
+		s.subscriptionMutex.RLock()
```

Comment:
Even on these examples where the immediate defer doesn't work because we're within a for-loop, I'd be tempted to refactor the body of the for-loop into a function to allow the unlock to be deferred (and because this method is probably scoring pretty high on cyclomatic complexity).